astropy/cosmology/core.py

```diff
@@ -94,8 +112,8 @@ class Cosmology(metaclass=abc.ABCMeta):
 
     # ---------------------------------------------------------------
 
-    def __init_subclass__(cls):
-        super().__init_subclass__()
+    def __init_subclass__(cls, *, skip_registration: bool = False):
+        _init_subclass(super().__init_subclass__, skip_registration)
```

Comment:
This function call seems very strange. Here, you do know that `skip_registration` is not used by the super class (and if anybody creates a new class in between, that's their problem), so this doesn't seem to be needed, i.e., why not have
```
super().__init_subclass__()
```